src/main/java/games/strategy/engine/framework/map/download/DownloadMapsWindow.java

```diff
@@ -116,14 +117,21 @@ private DownloadMapsWindow(
     if (!pendingDownloads.isEmpty()) {
       progressPanel.download(pendingDownloads);
     }
+    // TODO: there is a possibility that pendingDownloads will contain duplicates after the following call.
+    // i don't think it matters, but we should try to avoid it if possible. might be as simply as changing
+    // to a Set.
```

Comment:
This is a low-hanging fruit we can address in this PR, I think.  `DownloadFileDescription` implements `equals()` and `hashCode()`, so using a `Set` instead of a `List` should be no problem.  Can you please convert `pendingDownloads` to a `Set<DownloadFileDescription>` and pull the thread?  If it looks like it's going to require too many changes, let's just leave it as-is and get rid of the TODO, as it seems to be behaving correctly.